lasagne/layers/special.py

```diff
@@ -648,9 +658,25 @@ def __init__(self, incoming, localization_network, downsample_factor=1,
                              "output shape: (batch_size, num_input_channels, "
                              "input_rows, input_columns)")
 
+        if type(precompute_grid) is not bool and precompute_grid != 'auto':
+            raise ValueError("precompute_grid must be a boolean or set to "
+                             "'auto'.")
```

Comment:
I agree, I think that's clearer. Any other suggestions before I commit? I can rebase this one into a single commit if you're otherwise pleased with the PR.